### ISSUE TITLE
Redundant file system access to Delta's last transaction log file

### ISSUE THREAD

**Issue by findepi** [trinodb/trino#16780]

**Comment by ebyhr** [comment]
The reason for less access count of `00000000000000000000.json`~ `00000000000000000002.json` is cache in  `TransactionLogAccess.tableSnapshots`. If we disable the cache, these json files would be accessed 3 times as `00000000000000000003.json`. The 3 times access cause are the same as #16782

**Comment by findepi** [comment]
Yes. And that's expected behavior for the first getTableHandle call. But should not be done for eg split generation, otherwise we end up reading different state then the one we think we did.

**Comment by jkylling** [comment]
> Yes. And that's expected behavior for the first getTableHandle call. But should not be done for eg split generation, otherwise we end up reading different state then the one we think we did.

Indeed, we risk different state. The `loadSnapshot` method
`https://github.com/trinodb/trino/blob/16a730c7e9bd2b2f681016c8c9641c9c7b8f7ba6/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java#L150` could be augmented to take additional parameters `readVersion` and `checkpointReadVersion`, to ensure that we use the same table snapshot. This might also be a first step towards supporting time travel for Delta tables.

We also need a way to ensure that multiple references to the same Delta table in a query use the same snapshot version, to avoid https://github.com/jkylling/trino/commit/f8e32ed76d85ea5572f8cda43b5d6f3ccca706ab . Perhaps we can fix both types of issues at the same time.
